crates/storage/codecs/derive/src/lib.rs

```diff
@@ -51,15 +57,49 @@ mod compact;
 ///   efficient decoding.
 #[proc_macro_derive(Compact, attributes(maybe_zero, reth_codecs))]
 pub fn derive(input: TokenStream) -> TokenStream {
-    let is_zstd = false;
-    compact::derive(input, is_zstd)
+    compact::derive(input, None)
 }
 
 /// Adds `zstd` compression to derived [`Compact`].
-#[proc_macro_derive(CompactZstd, attributes(maybe_zero, reth_codecs))]
+#[proc_macro_derive(CompactZstd, attributes(maybe_zero, reth_codecs, reth_zstd))]
```

Comment:
adding reth_zstd makes sense, this a lot smarter than adding it to reth_codecs because then we don't run into the issue I had on my pr